views/_layout.pug

```diff
@@ -18,20 +18,12 @@ html(lang="en")
         include templates.pug
 
     block scripts
-        script(src=`${httpRootDir}client-options.js?hash=${optionsHash}`)
-        //-
-          The atob/base64 stuff below is to prevent any XSS attacks: now we have user content in the options (from
-          short links), it's possible to do XSS attacks, by leaving <script> tags in the source etc, which can be
-          interpreted by the browser.
+        script(src=`${httpRoot}client-options.js?hash=${optionsHash}`)
+        div#config(extraOptions=encodeURIComponent(compilerExplorerOptions), httpRoot=httpRoot, staticRoot=staticRoot)
```

Comment:
Maybe not a `<div>` here? Is there a non-visible tag we can put here just in case? or maybe just put `hidden` in here? Just in case some tiny bit of space is taken up.

Reply:
AFAIK the `div` should be zero size, but we could always move it to the `<head>` as `<meta>`?

Reply:
That works too; but as long as it doesn't show up. Maybe a meta tag, but don't block on this to merge :)